### PR TITLE
Accumulator for observables

### PR DESCRIPTION
**Pull request by KaiSzuttor**
Description of changes:
 -  added an accumulator for observable to get the mean and the variance.

### CODE REVIEW

**Comment by hmenke** [comment]
This should use `AutoParameters`.

**Comment by fweik** [comment]
To fix the boost error define BOOST_RESULT_OF_USE_DECLTYPE before including the boost headers. This is a boost bug that we have hit before (see Cells.hpp).

**Comment by fweik** [comment]
Looks good to me. Very terse. If you fix the boost issue, this can be merged.

File: src/core/accumulators/Accumulator.hpp [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+#define BOOST_RESULT_OF_USE_DECLTYPE
+#include <boost/accumulators/accumulators.hpp>
+#include <boost/accumulators/numeric/functional/vector.hpp>
+#include <boost/accumulators/statistics/mean.hpp>
+#include <boost/accumulators/statistics/stats.hpp>
+#include <boost/accumulators/statistics/variance.hpp>
+
+#include "observables/Observable.hpp"
+
+namespace ba = boost::accumulators;
```
Comment (fweik) [inline src/core/accumulators/Accumulator.hpp]: Please move this into the namespace, otherwise this has global visibility.

**Review by fweik — APPROVED** [review]

**Pull request merged**